### PR TITLE
Add more test cases to test_downsize

### PR DESCRIPTION
**Pull request by nathan-chan-brt**
## Description

We recently ran into an issue where skimage.transform.resize returned different results after upgrading from 0.18 to 0.19. Running these test cases convinced us it was more correct after the version upgrade. It is interesting to get this result though:
```
Original:
[[0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 1. 1. 0. 0. 0. 0. 0. 0. 0.]
 [0. 1. 1. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]]

Old resize:
[[0. 0. 0. 0. 0.]
 [1. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]]

New resize:
[[1. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0.]]
```

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

**Comment by stefanv** [comment]
I wonder if you were looking for https://scikit-image.org/docs/stable/api/skimage.transform.html#skimage.transform.downscale_local_mean?

Either way, I think more thorough testing is a good idea. Thank you!

**Comment by nathan-chan-brt** [comment]
> I wonder if you were looking for https://scikit-image.org/docs/stable/api/skimage.transform.html#skimage.transform.downscale_local_mean?

I don't think so? We need to be able to do nearest-neighbor interpolation.

**Review by lagru — APPROVED** [review]
Thanks @nathan-chan-brt. Always happy to add more test cases to our suite.

**Comment by nathan-chan-brt** [comment]
I'm not authorized to merge. Will someone please do it for me?

**Comment by lagru** [comment]
> I'm not authorized to merge. Will someone please do it for me?

In scikit-image we typically [require two approvals by maintainers before merging](https://scikit-image.org/docs/dev/skips/1-governance.html#decision-making-process). So with my approval the next maintainer who approves can immediately merge this but right now this approval is still missing. :slightly_smiling_face:

**Comment by stefanv** [comment]
Thanks @nathan-chan-brt!

**Pull request merged**
